streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreFactory.java

```diff
@@ -75,4 +76,79 @@ default void configure(final StreamsConfig config) {
 
     boolean isCompatibleWith(StoreFactory storeFactory);
 
+    class FactoryWrappingStoreBuilder<T extends StateStore> implements StoreBuilder<T> {
```

Comment:
This gave me the spins - previously the factories would wrap builders and suppliers so we could have a unified configuration interface. But now we're wrapping a factory in a builder, which is confusing and makes the code harder to reason about because when I have a builder instance it could actually be a factory underneath (not sure if that causes any problems, but it was nice that there was a non-circular hierarchy of wrapping before). Would it be better to instead revert the dsl operators to create processor instances that return unwrapped builders, and then wrap them in a factory from `addStateStore`? So, for example `KeyValueStoreMaterializer` would return a StoreBuilder like it used to, and then that gets wrapped by `addStateStore`?

Reply:
If this makes sense we could probably also clean this up after changing the rest of the DSL operators, since it would be hard to change `KeyValueStoreMaterializer` in the interim.

Reply:
as discussed offline, we are actively investigating how to disentangle the StoreFactory<->StoreBuilder wrapping mess. this is happening in parallel and doesn't need to block progress on the conversion PRs, which will rely on this FactoryWrappingStoreBuilder class for now